app/scripts/components/common/banner/index.tsx

```diff
@@ -1,78 +1,141 @@
 import React, { useState } from 'react';
-import { Icon } from '@trussworks/react-uswds';
+import { decode } from 'he';
 import {
   USWDSBanner,
-  USWDSBannerContent
+  USWDSBannerContent,
+  USWDSBannerButton,
+  USWDSBannerFlag,
+  USWDSBannerHeader,
+  USWDSBannerIcon,
+  USWDSBannerGuidance,
+  USWDSMediaBlockBody
 } from '$components/common/uswds/banner';
 
-const BANNER_KEY = 'dismissedBannerUrl';
-
-function hasExpired(expiryDatetime) {
-  const expiryDate = new Date(expiryDatetime);
-  const currentDate = new Date();
-  return !!(currentDate > expiryDate);
-}
-
-enum BannerType {
-  info = 'info',
-  warning = 'warning'
+interface GuidanceContent {
+  icon: string;
+  iconAlt?: string;
+  title: string;
+  text: string;
 }
 
-const infoTypeFlag = BannerType.info;
 interface BannerProps {
-  appTitle: string;
-  expires: Date;
-  url: string;
-  text: string;
-  type?: BannerType;
+  headerText?: string;
+  headerActionText?: string;
+  ariaLabel?: string;
+  flagImgAlt?: string;
+  leftGuidance?: GuidanceContent;
+  rightGuidance?: GuidanceContent;
+  className?: string;
+  defaultIsOpen?: boolean;
+  contentId?: string;
 }
 
+const defaultGuidance = {
+  left: {
+    title: 'Official websites use .gov',
+    text: 'A .gov website belongs to an official government organization in the United States.',
+    iconAlt: 'Dot gov icon',
+    icon: '/img/icon-dot-gov.svg'
+  },
+  right: {
+    title: 'Secure .gov websites use HTTPS',
+    text: `<>
+        A <strong>lock</strong> or <strong>https://</strong> means you've safely
+        connected to the .gov website. Share sensitive information only on
+        official, secure websites.
+      </>`,
+    iconAlt: 'HTTPS icon',
+    icon: '/img/icon-https.svg'
+  }
+};
+
 export default function Banner({
-  appTitle,
-  expires,
-  url,
-  text,
-  type = infoTypeFlag
+  headerText,
+  headerActionText = "Here's how you know",
+  ariaLabel,
+  flagImgAlt = '',
+  leftGuidance,
+  rightGuidance,
+  className = '',
+  defaultIsOpen = false,
+  contentId = 'gov-banner-content'
 }: BannerProps) {
+  const [isOpen, setIsOpen] = useState(defaultIsOpen);
 
-  const showBanner = localStorage.getItem(BANNER_KEY) !== url;
-  const [isOpen, setIsOpen] = useState(showBanner && !hasExpired(expires));
+  const defaultHeaderText =
+    'An official website of the United States government';
```

Comment:
⛏️ : headerActionText could probably follow this pattern too or headerText could follow headerActionText with a default value in props. If we keep it like this, this could probably be defined outside like `defaultGuidance` which would be beneficial too if we ever do test this to export for ref